Automattic/pocket-casts-android
# Add paging to clip sharing

## Description

This adds paging to the clip sharing page. There are some caveats that I'll explain.

### Pager indicator above the card

![Screenshot 2024-08-07 at 11 48 18](https://github.com/user-attachments/assets/386a35e6-002e-4de9-9ee2-c49463b63fae)

We decided to place the pager indicator above the card to accommodate small devices. Otherwise, users might be unaware that they can switch cards. There are other possible patterns, like showing a part of the next card peeking from an edge, but for now, we opted for the position change.

### Tablet layout issues

It doesn't look that great, and I don't know what to do about it. To support scrolling in a compact viewport, I need to set the `.weight()` modifier on scrollable content. This makes it expand to the maximum available space, but it inadvertently expands the content when there's too much space, such as on tablets.

![Screenshot 2024-08-07 at 09 26 15](https://github.com/user-attachments/assets/c7894315-f50d-423b-8dba-8487f17fddaa)

## Testing Instructions

1. Start clip sharing.
2. Check if paging works.
3. Tap on `Next`.
4. Paging should stop working.
5. Tap on `Edit clip`.
6. Paging should work again.

## Screenshots or Screencast 

![demo](https://github.com/user-attachments/assets/2120cb4b-d240-4dfc-9304-8e0ef8396238)

## Checklist
 
#### I have tested any UI changes...

- [ e] with the device set to have a large display and font size
